test/test_jit_py3.py

```diff
@@ -433,6 +433,32 @@ def foo():
                 return MyCoolNamedTuple(4, 5.5, [3])
             print(foo.graph)
 
+    def test_already_scripted_submodule(self):
+        class MyModel(nn.ModuleDict):
+            def __init__(self):
+                super().__init__()
+                self.layers: List[nn.Module] = []
+                for i in range(3):
+                    layer = torch.jit.script(nn.Linear(10, 10))
+                    self.add_module(str(i), layer)
+                    self.layers.append(layer)
+                self.layers = tuple(self.layers)
+
+            def forward(self, x):
+                for m in self.layers:
+                    x = m.forward(x)
```

Comment:
Note that the example we discussed offline has
```
x = m(x)
```
here instead. That does not work here because the `ConcreteType` generated for an `nn.Module` does not have a `__call__` attribute. That attribute is copied over to a `RecursiveScriptModule` instance from its eager `nn.Module` counterpart and things "just work" because it calls the scripted version of `forward`. It'd be great to fix this somehow but if not, I think that should be okay too? Unless there is a major difference between `m(x)` and `m.forward(x)`.

Reply:
Yeah that's ok for us.